hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -806,6 +878,225 @@ public Bytes getFailureResultFor(final ResponseCodeEnum status) {
 		}
 	}
 
+	/**
+	 * Executes the logic of creating a token from {@link HTSPrecompiledContract}.
+	 *
+	 * <p>When a token create call is received, the execution can follow one of the following 4 scenarios:</p>
+	 * <ol>
+	 *     <li>The calling smart contract has sent a corrupt input to the {@link HTSPrecompiledContract} (e.g.
+	 *     passing random bytes through the {@code encode()} Solidity method), which cannot be decoded to a valid
+	 *     {@link TokenCreateWrapper}.
+	 *     		<ul>
+	 *     		 	<li><b>result</b> - (TBD) the {@link DecodingFacade} throws an exception and null is returned
+	 *     		 	from the {@link HTSPrecompiledContract}</li>
+	 *     		 	<li><b>gas cost</b> - TBD</li>
+	 *     		</ul>
+	 *     	</li>
+	 *     <li>The decoding succeeds, we create a valid {@link TokenCreateWrapper}, but we cannot translate it to a
+	 *     valid token create {@link TransactionBody}. This comes from <b>difference in the design of the Solidity
+	 *     function interface and the HAPI API (protobufs)</b>
+	 *     		<ul>
+  	 *     		 	<li><b>result</b> - {@link MessageFrame}'s revertReason is set to the specific error and null is
+	 *     		 	returned from the {@link HTSPrecompiledContract}</li>
+	 * 	    		<li><b>gas cost</b> - TBD</li>
+	 *     		</ul>
+	 *     	</li>
+	 *     <li>The decoding succeeds, we create a valid {@link TokenCreateWrapper}, we successfully translate it to a
+	 *     valid token create {@link TransactionBody}. However, the {@link CreateChecks} validations find an input
+	 *     error.
+	 *     		<ul>
+	 *     		 	<li><b>result</b> - a child {@link ExpirableTxnRecord} is created, containing the error response
+	 *     		 	code. (from the point of view of the EVM this is a successful precompile call, however, from a
+	 *     		 	Hedera's perspective there has been a problem during the execution)
+	 *     		 	</li>
+	 * 	    		<li><b>gas cost</b> - TBD</li>
```

Comment:
The gas cost should be 100K, yes.

It appears that in a HAPI token create when the checks fail the hbars needed for the create still get charged, they are not refunded. 

As the implementation currently stands in the precompile, the behavior should be the same, since the logic for charging is executed before the checks, the balances are already updated at this point. We should add an E2E verifying this scenario to the test plan, though.